extractor/src/extract/query_builder.py

```diff
@@ -0,0 +1,159 @@
+from sqlalchemy import (
+    and_,
+    Column as AlchemyColumn,
+    Table,
+)
+from sqlalchemy.orm import (
+    aliased,
+    Query,
+)
+from typing import Callable, Dict
+
+from analyzer.src.analyze.attribute import Attribute
+from analyzer.src.analyze.sql_column import SqlColumn
+from extractor.src.config.service_logger import logger
+from extractor.src.errors import ImproperMappingError
+
+from arkhn_monitoring import Timer
+
+
+def handle_between_filter(col, value):
+    values = value.split(",")
+    if len(values) != 2:
+        raise ValueError("BETWEEN filter expects 2 values separated by a comma.")
+    min_val = values[0].strip()
+    max_val = values[1].strip()
+    return and_(col.__ge__(min_val), col.__le__(max_val))
+
+
+SQL_RELATIONS_TO_METHOD: Dict[str, Callable[[AlchemyColumn, str], Callable]] = {
+    "<": lambda col, value: col.__lt__(value),
+    "<=": lambda col, value: col.__le__(value),
+    "<>": lambda col, value: col.__ne__(value),
+    "=": lambda col, value: col.__eq__(value),
+    ">": lambda col, value: col.__gt__(value),
+    ">=": lambda col, value: col.__ge__(value),
+    "BETWEEN": handle_between_filter,
+    "IN": lambda col, value: col.in_(value.split(",")),
+    "LIKE": lambda col, value: col.like(value),
+}
+
+
+class QueryBuilder:
+    def __init__(self, extractor, analysis, pk_values):
```

Comment:
I think it's a mistake to make the QueryBuilder depend on the extractor, I would be far more comfortable if we passed the SqlAlchemy session and metadata independently